### PR TITLE
Theming

### PR DESCRIPTION
**Pull request by Simon-Laux**
- [X] Pull theme from file instead of saving it in local storage
- [x] Change Theme setting in settings
- Detect System Theme (Dark/Light) -> postponed (current approach (asking electron) doesn't work on my linux distro) 
- [X] Switch to a scss format based theme system
- [x] Update documentation
- ~make theme-name and -description (atleast of our official dc themes) translate-able~ for now english
- [x]  just load precompiled-css-themes and expand the tutorial on how to compile them in watch mode. (install node-sass globally and watch-compile the theme.)

this pr also includes some theming fixes.

Screenshot from 2020-05-04 07-25-33

### CODE REVIEW

File: src/main/themes.ts [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+export async function getAvailibleThemes(): Promise<Theme[]> {
+  // look at the common places for themes
+  return [
+    ...(await readThemeDir(dc_theme_dir, 'dc')),
+    ...(await readThemeDir(getCustomThemesPath(), 'custom')),
+  ]
+}
+
+export async function loadTheme(
+  theme_path: string
```
Comment (Jikstra) [inline src/main/themes.ts:72]: I'm against having a theme_path. I'm fine with having a theme name and two locations from where we load them. Either the installed ones, or from "~/.config/DeltaChat/themes". Everything else is a possible LFI vulnerability.

**Comment by Simon-Laux** [comment]
this function is internal and it isn't possible to load a theme from anywhere else than the two locations, read the docs file for on quick overview.

**Comment by Jikstra** [comment]
I get this error:
```
16.1s [D]main/deltachat/extras: theme system
16.1s [E]main/deltachat/extras: loading theme failed: theme system not found at: /home/kerle/workspace/deltachat-desktop/themes/t.scss 
Extras.<anonymous> (/home/kerle/workspace/deltachat-desktop/tsc-dist/main/deltachat/extras.js:37:21),
rejected (/home/kerle/workspace/deltachat-desktop/tsc-dist/main/deltachat/extras.js:6:65),
processTicksAndRejections (internal/process/task_queues.js:89:5)
```

File: bin/build/build-themes.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+          sass.render(
+            {
+              outputStyle: 'compressed',
+              data: theme_data,
+              includePaths: [dc_theme_dir],
+              sourceMap: false,
+            },
+            async (err, result) => {
+              if (err) {
+                rej(err)
```
Comment (Jikstra) [inline bin/build/build-themes.js:24]: `if (err) return rej(err)`

Then you don't need the else branch

File: src/renderer/ThemeManager.tsx [hunk 3]
```diff
@@ -1,89 +1,26 @@
...
-
-  getDefaultTheme() {
-    return defaultTheme
-  }
-}
+import { DeltaBackend } from './delta-remote'
+import { Theme } from '../shared/shared-types'
 
-export const manager = new ThemeManager()
+const { ipcRenderer } = window.electron_functions
```
Comment (Jikstra) [inline src/renderer/ThemeManager.tsx:4]: use ipcBackend here

**Pull request merged**